xapian-applications/omega/handler_poppler.cc

```diff
@@ -63,25 +65,25 @@ extract(const string& filename,
 	    return false;
 	}
 
-	int npages = doc->pages();
 	// Extracting PDF metadata
-	author = clear_text(doc->info_key("Author"));
-	title = clear_text(doc->info_key("Title"));
-	keywords = clear_text(doc->info_key("Keywords"));
+	author = text_to_utf8(doc->info_key("Author"));
+	title = text_to_utf8(doc->info_key("Title"));
+	keywords = text_to_utf8(doc->info_key("Keywords"));
+	int npages = doc->pages();
 	pages = str(npages);
 	// Extracting text from PDF file
 	for (int i = 0; i < npages; ++i) {
 	    page *p(doc->create_page(i));
 	    if (!p) {
-		cerr << "Poppler Error: Failed to create page" << endl;
-		return false;
+		cerr << "Poppler Error: Failed to create page " << i << endl;
+		continue;
```

Comment:
I'm not really clear about the change from `return false` to `continue` here - the commit message says "improve the error messages", but the actual changes also change the behaviour on error.  That makes me wonder if the commit message is incomplete, or the change to `continue` here an accidental one...

Under what circumstances might `create_page()` fail?

If (say) a page without text fails, then `continue` would definitely make sense.  If it's aiming to robustly handle a broken PDF file, then I guess that's OK, though there's also an argument for flagging up broken input.  Either way (or if this is for some other reason entirely) it'd be good to document the reason for this handling in a comment.

Also, what does pdftotext do in this situation?

Reply:
It seems that `pdftotext` return an error if `create_page()` returns NULL.
Actually, the problem is that if only one page fails and `extract` returns false, we lose all the information (including the metadata). I was trying to avoid that.
Maybe I should make this change in another commit(or in a new PR with a proper name).

Reply:
> Actually, the problem is that if only one page fails and `extract` returns false, we lose all the information (including the metadata). I was trying to avoid that.

But under what circumstances can that actually happen?

The poppler API documentation doesn't say what can cause `create_page()` to return `NULL`.  The handling inside poppler doesn't make it totally easy to see, but it looks to me like the list of pages which the file says it has would need to be inconsistent with the pages which are actually present, which sounds like a bad PDF file to me.

If `pdftotext` errors out in this case then that seems a sensible thing for us to do too, unless we have evidence that doing something else is actually better (such as some PDF files where `create_page()` fails for some pages, but which appear to be functional PDF files which you can view in a PDF viewer, etc).  We've also been feeding PDF files to `pdftotext` for years, so if such files exist we will already have been rejecting them for years.

If we are going to change this, it should really be a separate commit (the commit here already mixes up several changes - "use POPPLER_CFLAGS", "call clear_text() less", "improve error message" should ideally each have been their own commit).